### PR TITLE
[SPARK-7497] [PySpark] [Streaming] fix streaming flaky tests

### PR DESCRIPTION
**Pull request by davies**
Increase the duration and timeout in streaming python tests.

### CODE REVIEW

**Comment by tdas** [comment]
test this again

**Comment by andrewor14** [comment]
retest this please

**Comment by tdas** [comment]
May be a good idea to merge this any way to reduce some test flakiness.

File: python/pyspark/streaming/tests.py [hunk 1]
```diff
@@ -41,8 +41,8 @@
 
 class PySparkStreamingTestCase(unittest.TestCase):
 
-    timeout = 4  # seconds
-    duration = .2
```
Comment (tdas) [inline python/pyspark/streaming/tests.py]: What is the duration? Batch duration? Why increase this?

**Comment by tdas** [comment]
@davies  Why are you increasing the batch duration??

**Comment by davies** [comment]
@tdas The tests become flaky with lower duration.

**Comment by tdas** [comment]
Could you explain why does it become flaky? I want to understand it so that
we can learn some lessons that may help on the Scala/Java side as well.

**Comment by davies** [comment]
@tdas Can not reproduce this flakiness locally right now. Remembered that when the duration is low and the job can not finished within it, something weird happened few months ago.

**Comment by tdas** [comment]
@JoshRosen Flaky test?
org.apache.spark.SparkContextSuite.Cancelling job group should not cause SparkContext to shutdown (SPARK-6414)

File: python/pyspark/streaming/tests.py (hunk 1 above)
Comment (davies) [inline python/pyspark/streaming/tests.py]: It's batch duration. These tests became flaky when we change the duration from 1 to 0.2, I'd like to try  to change it to 0.5. If they are still flaky, then move to 1 second.

**Comment by tdas** [comment]
The problem with this approach is that it will increase the run time of python streaming tests by a lot. And since these tests run 3 times for 3 different version of python, the increase will be multiple. But probably that is okay for now. The flakiness is causing problems for others, so I will merge this.

**Comment by tdas** [comment]
I have merged it master and branch-1.4

**Pull request closed without merging**